lib/cstruct.ml

```diff
@@ -96,7 +96,7 @@ let create_unsafe len =
   { buffer ; len ; off = 0 }
 
 let check_bounds t len =
-  Bigarray.Array1.dim t.buffer >= len
+  len >= 0 && Bigarray.Array1.dim t.buffer >= len
```

Comment:
I don't think check_bounds needs to be public any more. It was originally because it was inserted by a camlp4 extension, but nowadays it's just done inline in the library itself.